pkg/sac/scope_checker_or.go

```diff
@@ -0,0 +1,212 @@
+package sac
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/auth/permissions"
+	"github.com/stackrox/rox/pkg/sac/effectiveaccessscope"
+)
+
+type orScopeChecker struct {
+	scopeCheckers []ScopeChecker
+}
+
+// NewOrScopeChecker returns a new scope checker wrapping the given scope checkers,
+// each result will be combined in a disjunction (OR).
+func NewOrScopeChecker(scopeCheckers ...ScopeChecker) ScopeChecker {
+	return orScopeChecker{
+		scopeCheckers: scopeCheckers,
+	}
+}
+
+func (s orScopeChecker) Core() ScopeCheckerCore {
+	if len(s.scopeCheckers) != 0 {
+		return s.scopeCheckers[0].Core()
+	}
+	return nil
+}
+
+func (s orScopeChecker) SubScopeChecker(keys ...ScopeKey) ScopeChecker {
+	for i := range s.scopeCheckers {
+		s.scopeCheckers[i] = s.scopeCheckers[i].SubScopeChecker(keys...)
+	}
+	return s
+}
+
+func (s orScopeChecker) PerformChecks(ctx context.Context) error {
+	var performChecksErrs *multierror.Error
+	for _, checker := range s.scopeCheckers {
+		err := checker.PerformChecks(ctx)
+		// Short-circuit on the first non-error perform check.
+		if err != nil {
+			performChecksErrs = multierror.Append(performChecksErrs, err)
+		}
+	}
+	if performChecksErrs == nil {
```

Comment:
Why this extra `if`-block?

Reply:
Weirdly enough, I'm not sure this is *multierror.Error specific, when `performChecksErrs` is nil, the unit test will fail and detect an error. Hence the explicit check for `nil` here, instead of simply returning.

Reply:
`.ErrorOrNil()` should help : ).